app/models/user.rb

```diff
@@ -44,10 +40,12 @@ def email=(email)
   #
   # @return [UserEmail] The user's primary email address record.
   def default_email_record
-    valid_emails = emails.each.select { |email_record| !email_record.marked_for_destruction? }
-    result = valid_emails.find { |email_record| email_record.primary? }
+    valid_emails = emails.each.select { |email_record| !email_record.marked_for_destruction? }###########
+    result = valid_emails.detect(&:primary?)
     result ||= valid_emails.first
     result ||= emails.order('primary=1 ASC').first
+
+    [].reduce([]) {|_|}
```

Comment:
Prefer `reduce` over `reduce`.<br>Name `reduce` block params `|a, e|`.<br>Space between { and | missing.<br>Space missing inside }.